apps/web/pages/auth/login.tsx

```diff
@@ -211,6 +224,40 @@ const _getServerSideProps = async function getServerSideProps(context: GetServer
   const session = await getSession({ req });
   const ssr = await ssrInit(context);
 
+  const verifyJwt = (jwt: string) => {
+    const secret = new TextEncoder().encode(process.env.CALENDSO_ENCRYPTION_KEY);
+
+    return jwtVerify(jwt, secret, {
+      issuer: WEBSITE_URL,
+      audience: `${WEBSITE_URL}/auth/login`,
+      algorithms: ["HS256"],
+    });
+  };
+
+  let totpEmail = null;
+  if (context.query.totp) {
+    try {
+      const decryptedJwt = await verifyJwt(context.query.totp as string);
+      if (decryptedJwt.payload) {
+        totpEmail = decryptedJwt.payload.email as string;
+      } else {
+        return {
+          redirect: {
+            destination: "/auth/error?error=JWT%20Invalid%20Payload",
```

Comment:
If we are in the presence of a totp JWT token and we don't get a payload, we will redirect the user to an auth error "JWT Invalid Payload"